File: articles/jung-freud-and-the-divergence-of-depth-psychologies/fortran/depth_score.f90
! Jung, Freud, and the Divergence of Depth Psychologies
! Fortran example: simple synthetic depth-psychology comparison score

program depth_score_demo
  implicit none

  real :: f, j

  f = 0.66 * 0.84 + 0.70 * 0.78 + 0.62 * 0.82 + &
      0.58 * 0.76 + 0.50 * 0.69 - 0.30 * 0.28

  j = 0.58 * 0.36 + 0.70 * 0.30 + 0.62 * 0.34 + &
      0.64 * 0.28 + 0.60 * 0.38 + 0.56 * 0.32 - 0.24 * 0.84

  print *, "Synthetic Freudian score:", f
  print *, "Synthetic Jungian score:", j
end program depth_score_demo

File: articles/jung-freud-and-the-divergence-of-depth-psychologies/c/depth_score.c
/*
Jung, Freud, and the Divergence of Depth Psychologies
C example: simple synthetic depth-psychology comparison score

Synthetic educational demonstration only.
Not for diagnosis, assessment, screening, surveillance, or individual evaluation.
*/

#include <stdio.h>

double freudian_score(
    double repression,
    double sexuality_conflict,
    double infantile_history,
    double defense_intensity,
    double transference_pressure,
    double mythic_amplification
) {
    return 0.66 * repression
         + 0.70 * sexuality_conflict
         + 0.62 * infantile_history
         + 0.58 * defense_intensity
         + 0.50 * transference_pressure
         - 0.30 * mythic_amplification;
}

double jungian_score(
    double compensation,
    double archetypal_density,
    double prospective_development,
    double mythic_amplification,
    double symbolic_coherence,
    double individuation_pressure,
    double repression
) {
    return 0.58 * compensation
         + 0.70 * archetypal_density
         + 0.62 * prospective_development
         + 0.64 * mythic_amplification
         + 0.60 * symbolic_coherence
         + 0.56 * individuation_pressure
         - 0.24 * repression;
}

int main(void) {
    double f = freudian_score(0.84, 0.78, 0.82, 0.76, 0.69, 0.28);
    double j = jungian_score(0.36, 0.30, 0.34, 0.28, 0.38, 0.32, 0.84);

    printf("Synthetic Freudian score: %.3f\n", f);
    printf("Synthetic Jungian score: %.3f\n", j);
    return 0;
}
